File: src/main/java/problems/easy/MajorityElement/MajorityElement.scala
package problems.easy.MajorityElement

object SolutionScala {
  def majorityElement(nums: Array[Int]): Int = {
    val minNumberNeeded = math.floor(nums.length / 2) + 1
    var results: scala.collection.mutable.Map[Int, Int] =
      scala.collection.mutable.Map()
    for (i <- nums.indices) {
      if (!results.contains(nums(i))) {
        results += (nums(i) -> 1)
      } else {
        results(nums(i)) = results(nums(i)) + 1
      }
    }
    println(results)
    for ((k, v) <- results) {
      println(s"key = {k}, values = {v}")
      if (v >= minNumberNeeded) {
        return k
      }
    }

    0
  }
}

File: src/main/java/problems/easy/MajorityElement/majority_element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
  let minNumberNeeded = Math.floor(nums.length / 2) + 1;
  let results = new Map();
  for (let i = 0; i < nums.length; i++) {
    if (results.has(nums[i])) {
      results.set(nums[i], results.get(nums[i]) + 1);
    } else {
      results.set(nums[i], 1);
    }
  }
  for (let [key, value] of results) {
    if (value >= minNumberNeeded) {
      return key;
    }
  }
  return 0;
};

module.exports = { majorityElement };
